deeplinkdispatch-processor/src/test/java/com/airbnb/deeplinkdispatch/BaseDeepLinkProcessorTest.kt

```diff
@@ -0,0 +1,140 @@
+                    .containsExactlyInAnyOrder(*generatedFiles.keys.toTypedArray())
+                generatedFiles.keys.forEach { filename ->
+                    Assertions.assertThat(result.generatedFiles[filename]?.readText())
+                        .isEqualTo(generatedFiles[filename])
+                }
+                // Ksp generated files do not compile in tests so so not try to load them.
+                if (!result.useKsp) {
+                    val generatedRegistryClazz =
+                        result.result.classLoader.loadClass(registryClassName)
+                    val baseRegistryClazz =
+                        result.result.classLoader.loadClass("com.airbnb.deeplinkdispatch.BaseRegistry")
+                    Assertions.assertThat(generatedRegistryClazz).hasDeclaredMethods("matchIndex0")
+                    val registryInstance = generatedRegistryClazz.newInstance()
+                    Assertions.assertThat(registryInstance).isNotNull
+                    Assertions.assertThat(
+                        baseRegistryClazz.getDeclaredMethod("getAllEntries")
+                            .invoke(registryInstance) as List<DeepLinkEntry>
+                    ).isEqualTo(
+                        indexEntries
+                    )
+                }
+            }
+        }
+
+        internal fun assertCompileError(
+            results: List<CompileResult>,
+            errorMessage: String
+        ) {
+            results.forEach { result ->
+                Assertions.assertThat(result.result.exitCode)
+                    .isEqualTo(KotlinCompilation.ExitCode.COMPILATION_ERROR)
+                Assertions.assertThat(result.result.messages).contains(
+                    errorMessage
+                )
+            }
+        }
+
+        internal fun compile(
+            sourceFiles: List<SourceFile>,
+            arguments: MutableMap<OptionName, OptionValue>? = null,
+            useKsp: Boolean = false
+        ): CompileResult {
+            val compilation = KotlinCompilation().apply {
+                sources = sourceFiles
+                if (useKsp) {
+                    symbolProcessorProviders = listOf(DeepLinkProcessorProvider())
+                    arguments?.let { kspArgs = arguments }
+                } else {
+                    annotationProcessors = listOf(DeepLinkProcessor())
+                    arguments?.let { kaptArgs = arguments }
+                }
+                inheritClassPath = true
+                messageOutputStream = System.out
+            }
+            val result = compilation.compile()
+            val generatedSources = if (useKsp) {
+                compilation.kspSourcesDir.walk().filter { it.isFile }.toList()
+            } else {
+                result.sourcesGeneratedByAnnotationProcessor
+            }
+            return CompileResult(result, generatedSources.map { it.name to it }.toMap(), useKsp)
+        }
+
+        internal fun compileIncremental(
```

Comment:
This name makes me think we are going to actually test incremental compilation (which I believe is possible with this compile testing library). I think this is another example of code cleanups we could get by removing the old `*` mode.